kernel/thread.c

```diff
@@ -680,6 +687,13 @@ static long _syscall_clone_vfork(
         child->run_thread = myst_run_thread;
         child->main.thread_group_lock = MYST_SPINLOCK_INITIALIZER;
         child->thread_lock = &child->main.thread_group_lock;
+        {
+            char buf[BUFSIZ];
+            /* ATTN: we don't take a lock on _num_threads,
+             thread names could be duplicates */
+            snprintf(buf, sizeof(buf), "thread-%ld", _num_threads);
+            myst_set_thread_name(child, buf);
```

Comment:
Why don't we just sprintf into the myst_thread_t name buffer?

Reply:
Probably a theoritical concern, but `_num_threads` being a unsigned long int, could cause the thread name to go over 16 bytes. So this was delegating to `myst_set_thread_name`, which does the bounds checking anyways.

Reply:
We can check `_num_threads` here. Checking an integer is always faster than checking a string.

Reply:
Changed to not use `myst_set_thread_name`.